shell/platform/glfw/flutter_glfw.cc

```diff
@@ -521,6 +521,13 @@ static FLUTTER_API_SYMBOL(FlutterEngine)
     config.open_gl.make_resource_current = GLFWMakeResourceContextCurrent;
     config.open_gl.gl_proc_resolver = GLFWProcResolver;
   }
+  // Provide path to .so in AOT configurations.
+  if (FlutterEngineRunsAOTCompiledDartCode()) {
```

Comment:
As mentioned in the documentation, the command line args are meant to only access [unstable features](https://github.com/flutter/engine/blob/2b07b71ad01456308e152d19121c024e6aef97fe/shell/platform/embedder/embedder.h#L847) and the flags themselves are not guaranteed to stay consistent. The [4 pointers to the isolate and instructions](https://github.com/flutter/engine/blob/2b07b71ad01456308e152d19121c024e6aef97fe/shell/platform/embedder/embedder.h#L1469) mappings on the other hand are stable. You may dlopen and dlsym the same from this `libapp.so`.

It is entirely reasonable that you think that this is too complicated in lieu of just accessing features via a unstable interface. In that case, we can expose another field in the `FlutterProjectArgs` that exposes a placeholder for callers to specify a library containing the 4 asset blobs.

Reply:
Yeah, If the engine can already do this work it seems ... unnecessary. But, this isn't a high priority item, so I'd rather not take on the additional work here.

Reply:
Thats fair. Tracking that work item in https://github.com/flutter/flutter/issues/43454. Thanks for looking into this.